src/main/java/com/singh/astha/medicinereminder/dtos/transformers/DosageDtoTransformer.java

```diff
@@ -5,11 +5,22 @@
 import com.singh.astha.medicinereminder.dtos.ResponseDto.MedicineResponseDto;
 import com.singh.astha.medicinereminder.enums.DosageType;
 import com.singh.astha.medicinereminder.models.DosageHistory;
+import com.singh.astha.medicinereminder.models.Medicine;
 import org.springframework.stereotype.Component;
 
 @Component
 public class DosageDtoTransformer {
 
+    public static DosageHistory setDosageHistory(Long userId, Medicine savedMedicine) {
+        DosageHistory dosageHistory = new DosageHistory();
+        dosageHistory.setMedicineId(savedMedicine.getId());
+        dosageHistory.setMedicine(savedMedicine);
+        dosageHistory.setDosage(savedMedicine.getCurrentDosage());
+        dosageHistory.setType(DosageType.REFILL);
+        dosageHistory.setUserId(userId);
```

Comment:
This method shouldn't be a transformer.

Create a private method in the same call and call it as initialiseDosageHistory